plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/ExMemory.hs

```diff
@@ -94,13 +96,36 @@ type CostingInteger =
     SatInt
 #endif
 
-
 -- $(if finiteBitSize (0::SatInt) < 64 then [t|Integer|] else [t|SatInt|])
 
+{- | Divide one costing integer by another, "rounding upwards".  This is needed
+when we expose costs to the ledger, which will use different (smaller) units.
+Suppose one ledger unit is 1000 real cost units: then if a script costs 12345678
+real units we want that to be converted to 12346 ledger units, since 12345 (as
+given by `div`) would convert to 12345000 real units, which wouldn't be enough
+to run the script.  We want a <= (a `uppperIntegerQuotient` b) * b < a+b for all
+a and for all b>0, except that we may get '==' instead of '<' when we're using
+SatInt and a+b == MaxBound.  The behaviour when b <= 0 is unspecified.
+-}
+uppperIntegerQuotient :: CostingInteger -> CostingInteger -> CostingInteger
```

Comment:
We don't say what happens if b<=0 (and what happens isn't good), so should we have an error/exception here if that's the case?

Reply:
The problem is what happens if b<0: it's not clear to me what the result should be in that case (and why would you be scaling times by a negative amount anyway?), so I'm tempted just to exclude that entirely, and then we might as well exclude zero as well.

Reply:
Typo: ppp

Reply:
Hah!  I managed to do that everywhere without noticing.  I must have been relying on emacs to fill in the big name for me.